cpp/arcticdb/pipeline/read_frame.cpp

```diff
@@ -338,8 +352,8 @@ void decode_into_frame_static(
                         it.dest_col(),
                         m.frame_field_descriptor_.name());
 
-            util::check(data != end, "Reached end of input block with {} fields to decode", it.remaining_fields());
-            decode_or_expand(data, buffer.data() + m.offset_bytes_, encoded_field, m.source_type_desc_,  m.dest_bytes_, buffers);
+            util::check(data != end || remaining_fields_empty(it, context), "Reached end of input block with {} fields to decode", it.remaining_fields());
```

Comment:
I've seen that they were not used at all in the current code and found that a bit confusing. I didn't know that we'll need them for the arrays. If it's OK I'll leave it as is and add them again when we need them.

Reply:
I don't think that a comment will add any value. Moreover that nothing in the code uses the buffers param. It's created in one of the top-level functions and passed in each nested function. TBH I'm not sure this is the best way to handle this. Maybe it would be better adding some sort of global buffer manager singleton, I don't know. My point is that we're not 100% sure what shape will this take, so adding a comment "We will add x, y, z here" might raise more questions than it answers.

Reply:
On reflection, I'm not convinced that moving the type handling changes in piecemeal has been a huge success, given that it already exists in its entirety in BitBucket. Small PRs are nice, but IMO they should be as small as they can be to still make sense, and no smaller. Maybe something to consider for future planning.

Reply:
I have reverted the changes and brought the buffers back.